client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,31 +1,183 @@
 import { Title } from '@automattic/onboarding';
 import { Plans2023Tooltip, useManageTooltipToggle } from '@automattic/plans-grid-next';
-import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
 import React, { useState, useEffect } from 'react';
 import ButtonGroup from 'calypso/components/button-group';
-import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import { useSelectedPlanUpgradeMutation } from 'calypso/data/import-flow/use-selected-plan-upgrade';
-import { useSelector } from 'calypso/state';
-import { getSitePlan, getSitePlanRawPrice } from 'calypso/state/sites/plans/selectors';
 import { UpgradePlanFeatureList } from './upgrade-plan-feature-list';
 import { UpgradePlanHostingDetails } from './upgrade-plan-hosting-details';
 
+interface SwitcherProps {
+	selectedPlan: string;
+	onMonthlyPlanClick: () => void;
+	onAnnualPlanClick: () => void;
+}
+
+const UpgradePlanPeriodSwitcher = ( props: SwitcherProps ) => {
+	const translate = useTranslate();
+
+	if ( config.isEnabled( 'migration-flow/introductory-offer' ) ) {
+		return null;
+	}
+
+	const { selectedPlan, onMonthlyPlanClick, onAnnualPlanClick } = props;
+
+	return (
+		<div className="import__upgrade-plan-period-switcher">
+			<ButtonGroup>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS_MONTHLY } ) }
+					onClick={ onMonthlyPlanClick }
+				>
+					{ translate( 'Pay monthly' ) }
+				</Button>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS } ) }
+					onClick={ onAnnualPlanClick }
+				>
+					{ translate( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+interface PlanPriceOfferProps {
+	plan?: Plan;
+	currencyCode?: string;
+	originalMonthlyPrice?: number;
+	introOfferMonthlyPrice?: number;
+	originalFullPrice?: number;
+	introOfferFullPrice?: number;
+	introOfferEnabled: boolean;
+	isIntroOfferComplete: boolean;
+}
+
+const PlanPriceOffer = ( props: PlanPriceOfferProps ) => {
+	const translate = useTranslate();
+
+	const {
+		plan,
+		originalFullPrice,
+		introOfferFullPrice,
+		introOfferEnabled,
+		isIntroOfferComplete,
+		introOfferMonthlyPrice,
+		originalMonthlyPrice,
+		currencyCode,
+	} = props;
+
+	const showIntroOffer =
+		introOfferEnabled &&
+		! isIntroOfferComplete &&
+		introOfferMonthlyPrice &&
+		originalMonthlyPrice &&
+		introOfferFullPrice &&
+		originalFullPrice &&
+		currencyCode;
+
+	if ( ! showIntroOffer ) {
+		return (
+			<div className="import__upgrade-plan-price">
+				<PlanPrice rawPrice={ originalMonthlyPrice } currencyCode={ currencyCode } isSmallestUnit />
+				<span className="plan-time-frame">
+					<small>{ plan?.getBillingTimeFrame() }</small>
+				</span>
+			</div>
+		);
+	}
+
+	const billingTimeFrame = translate(
+		'per month, %(discountedPrice)s billed annually for the first year, %(originalPrice)s afterwards, excl. taxes',
```

Comment:
Minor thought: it may be clearer to specify `per year` for the final recurring price.
```suggestion
		'per month, %(discountedPrice)s billed annually for the first year, %(originalPrice)s per year afterwards, excl. taxes',
```